lib/Interpreter/IncrementalParser.cpp

```diff
@@ -125,7 +125,7 @@ namespace {
   if (!II)
     return;
   const clang::DefMacroDirective* MD
-    = llvm::dyn_cast<clang::DefMacroDirective>(PP.getMacroDirective(II));
+    = llvm::dyn_cast_or_null<clang::DefMacroDirective>(PP.getMacroDirective(II));
```

Comment:
On 15/06/16 21:06, Axel Naumann wrote:
>
> In lib/Interpreter/IncrementalParser.cpp 
> <https://github.com/root-mirror/cling/pull/17#discussion_r67225773>:
>
> > @@ -125,7 +125,7 @@ namespace {
> >    if (!II)
> >      return;
> >    const clang::DefMacroDirective* MD
> > -    = llvm::dyn_cast<clang::DefMacroDirective>(PP.getMacroDirective(II));
> > +    = llvm::dyn_cast_or_null<clang::DefMacroDirective>(PP.getMacroDirective(II));
>
> @marsupial <https://github.com/marsupial> is suggesting the correct 
> fix. We need a cling test with -x c :-)
>
Could you provide a test with this fix?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/root-mirror/cling/pull/17/files/252d215c9d8be48c614e85447f52951f37135ec5..6ac37ef57b76f3b4469856f12ce069b03a57c678#r67225773>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AGNuU36msdl2rBQgRWybReBwUtVnVqdEks5qME1BgaJpZM4I0etZ>.
>